sdk/src/lib.rs

```diff
@@ -206,11 +210,18 @@ impl Worker {
         );
     }
 
+    /// Insert Custom App Context for Workflows and Activities
+    pub fn insert_app_data<T: Send + Sync + 'static>(&mut self, data: T) {
+        (*Arc::get_mut(&mut self.app_data)
+            .expect("app_data can only be inserted during initial setup"))
+        .insert(data);
```

Comment:
What would probably make sense to do here is not `Arc` this data in `Worker`, but only once `run` has started, that way you can avoid this expectation. It could be behind an `Option` and is `take`n when run starts, and replaced when it's done.

Reply:
done.